applications: asset_tracker_v2: Use Location library (#8631)
nrfconnect/sdk-nrf: doc/nrf/releases/release-notes-changelog.rst

* Updated:

  * The application now uses the new LwM2M location assistance objects through the :ref:`lib_lwm2m_location_assistance` library.
  * The application now uses the :ref:`lib_location` library for retrieving location information.

>     This is a major change in the application code although much smaller change from user perspective.

Reviewer: ```suggestion
    This is a major change in the application code although much smaller change from the user perspective.
```

    Here are additional notes about the change:
    * Added location module and removed GNSS module. GNSS used through Location library.
    * Neighbor cell handling moved from modem module to location module to be used through Location library
    * Location module triggers location_request() with GNSS being 1st priority method and cellular 2nd priority if they are enabled in application configuration
    * NMEA support removed. This was there only because nRF Cloud originally didn't support PVT but that has changed.
    * AGPS/PGPS are not requested based on triggers in the application but only based on Location library events LOCATION_EVT_GNSS_ASSISTANCE_REQUEST and LOCATION_EVT_GNSS_PREDICTION_REQUEST

>     * LTE LC neighbor cell search type cannot be configured or defined with Location library at the moment. Default search type is always.

Reviewer: ```suggestion
    * LTE LC neighbor cell search type cannot be configured or defined with Location library at the moment. Default search type is "always".
```

Author: This sentence seems unfinished. I think it should have been:
`Default search type is always used.`
I've now modified it.

Reviewer: Ah, ok!

>     * The Kconfig option ``CONFIG_GNSS_MODULE_PGPS_STORE_LOCATION`` (calling ``nrf_cloud_pgps_set_location()``) is not supported in the Location library.

Reviewer 2: Not sure if we should mention this option now that it's removed

Author: I thought we are supposed to mention if we remove some functionality. @peknis @simensrostad What to do?

Reviewer 2: Right, I'll defer to @peknis

  * The application now uses the new LwM2M location assistance objects through the :ref:`lib_lwm2m_location_assistance` library.
  * Added handling for the new data receive events in the :ref:`lib_nrf_cloud` library.
  * The application now uses the :ref:`lib_location` library for retrieving location information.
    This is a major change in the application code but a minor change from the user perspective.

>     * Added :ref:`asset_tracker_v2_location_module` and removed GNSS module. GNSS used through Location library.

Reviewer 3: ```suggestion
    * Added :ref:`asset_tracker_v2_location_module` and removed GNSS module.
      GNSS is used through the Location library.
```

    * Neighbor cell handling moved from :ref:`asset_tracker_v2_modem_module` to :ref:`asset_tracker_v2_location_module` to be used through Location library
    * The :ref:`asset_tracker_v2_location_module` triggers a ``location_request()`` with GNSS being the first priority method and cellular the second priority if they are enabled in the application configuration

>     * NMEA support removed. This was there only because originally, nRF Cloud did not support PVT but that has changed.

Reviewer 3: ```suggestion
    * NMEA support removed.
      This was there only because originally, nRF Cloud did not support PVT but that has changed.
```

>     * AGPS/PGPS are not requested based on triggers in the application but only based on Location library events LOCATION_EVT_GNSS_ASSISTANCE_REQUEST and LOCATION_EVT_GNSS_PREDICTION_REQUEST

Reviewer 3: ```suggestion
    * A-GPS/P-GPS are not requested based on triggers in the application but only based on Location library events ``LOCATION_EVT_GNSS_ASSISTANCE_REQUEST`` and ``LOCATION_EVT_GNSS_PREDICTION_REQUEST``.
```

>     * Currently, you cannot configure or define the LTE LC neighbor cell search type with the Location library. The default search type is always used.

Reviewer 3: ```suggestion
    * Currently, you cannot configure or define the LTE LC neighbor cell search type with the Location library.
      The default search type is always used.
```

>     * The Kconfig option ``CONFIG_GNSS_MODULE_PGPS_STORE_LOCATION`` (calling ``nrf_cloud_pgps_set_location()``) is not supported in the Location library.

Reviewer 3: ```suggestion
    * The Kconfig option :kconfig:option:`CONFIG_GNSS_MODULE_PGPS_STORE_LOCATION` (calling :c:func:`nrf_cloud_pgps_set_location()`) is not supported in the Location library.
```

* :ref:`lib_location` library:

  * Updated:
    * Added timeout for the entire location request.
    * Added location data details such as entire PVT data.

>     * Moved location method from :c:struct:`location_data` to :c:struct:`location_event_data`.

Reviewer 3: ```suggestion
    * Moved location method from the :c:struct:`location_data` structure to :c:struct:`location_event_data`.
```
